### DISCUSSION TITLE
Training with older v0 versus new v1

### DISCUSSION THREAD

**Discussion by AsmoKoskinen** [SWivid/F5-TTS#867] [Q&A] [answered]
"F5-TTS v1 base model with better training and inference performance."

I trained a Finnish model at home (4070 Ti Super (16GB)). I used about four days of Finnish speeches as my data (...total duration of all wav files: 106:10:05.17 (HH:MM:SS.ss)). The training took approximately four days.

What are the benefits of retraining the model with v1? I might buy a 5090 (32G) one day, but I have no plans to train before that.

![Kuvakaappaus%20-%202024-12-17%2014-10-00](https://github.com/user-attachments/assets/c724b2b0-d049-432e-b1fd-b909537c1826)

**Comment by SWivid** [comment, marked as answer]
> What are the benefits of retraining the model with v1?

"F5-TTS v1 base model with better training and inference performance." is in comparison with v0.
If you use same setting (learning rate, max updates, total batch size), v1 converges faster and gives better WER/SIM result (lower word error rate, higher speaker similarity).

**Reply by AsmoKoskinen** [reply]
Thank you, when I got some day faster GPU I do that again.

**Reply by AsmoKoskinen** [reply]
I started a new training with the same Finnish dataset as before. Looks good for the first day. And sounds, too.

RTX 4070 Ti Super.

{
    "exp_name": "F5TTS_v1_Base",
    "learning_rate": 0.0001,
    "batch_size_per_gpu": 2000,
    "batch_size_type": "frame",
    "max_samples": 96,
    "grad_accumulation_steps": 16,
    "max_grad_norm": 0.3,
    "epochs": 200,
    "num_warmup_updates": 3000,
    "save_per_updates": 10000,
    "keep_last_n_checkpoints": -1,
    "last_per_updates": 5000,
    "finetune": true,
    "file_checkpoint_train": "",
    "tokenizer_type": "pinyin",
    "tokenizer_file": "",
    "mixed_precision": "bf16",
    "logger": "tensorboard",
    "bnb_optimizer": true
}

---

![Kuvakaappaus - 2025-03-21 10-43-39](https://github.com/user-attachments/assets/ed726861-d187-49ac-a44a-d4997238d187)

---

![Kuvakaappaus - 2025-03-21 10-43-59](https://github.com/user-attachments/assets/e1f46226-0827-4f9f-93e8-e5b505b3cf33)

---

![Kuvakaappaus - 2025-03-21 10-43-50](https://github.com/user-attachments/assets/cbc6ec06-8e43-486c-8841-6c70d594ce34)

---

**Reply by AsmoKoskinen** [reply]
Done. Three days, 153 epochs and 195 000 steps. Dataset was 106 hours.
---
![Kuvakaappaus - 2025-03-23 10-08-20](https://github.com/user-attachments/assets/f97520c2-10a5-43c3-9940-1274dce70bb5)
---
![Kuvakaappaus - 2025-03-23 10-08-55](https://github.com/user-attachments/assets/c167588d-5a0c-48b2-83a6-69e2e0d570bf)
---
![Kuvakaappaus - 2025-03-23 10-09-04](https://github.com/user-attachments/assets/74b3bc79-f7d2-4401-b81c-cbbb065c9e8e)
---
![Kuvakaappaus - 2025-03-23 10-09-24](https://github.com/user-attachments/assets/6b10689d-186f-42ea-b7f8-8c141191486e)
---
![Kuvakaappaus - 2025-03-23 10-09-42](https://github.com/user-attachments/assets/4fe2ac44-a81d-44ca-91f6-3466f7208f95)
---

**Comment by AsmoKoskinen** [comment]
Claude says:

**Summary of the Training Process**

The training has progressed excellently and achieved its goals! Here's a summary of the final screenshots:

_Training Status_

Progress: 153/200 epochs (76.5% complete)
Total time: 3 days 0 hours 17 minutes
Step count: 195.1k
Current loss: ~0.627 (smoothed), individual values vary

_Loss Curve Analysis_

Loss has decreased steadily throughout the entire training from the initial ~0.67 to the current ~0.627
During the last day, the loss curve shows a particularly clear downward trend
The curve shows no signs of plateauing, suggesting the model could benefit from additional training

_Individual Loss Values_

The terminal view shows excellent individual values:

0.441, 0.466, 0.496, 0.497

These are significantly better than at the beginning of training

_Conclusions_

1. The training has been highly successful

2. The model's performance has improved consistently throughout the training

3. Reducing the learning rate was a good decision as it helped the model find lower loss values

4. The model might benefit from additional training as the loss curve continues to decline

I recommend testing different checkpoints (e.g., latest vs. epoch 100) to compare audio quality. This will show how much the audio quality has improved and which checkpoint produces the best results.

**Reply by AsmoKoskinen** [reply]
First two days:

{
    "exp_name": "F5TTS_v1_Base",
    "learning_rate": 0.0001,
    "batch_size_per_gpu": 2000,
    "batch_size_type": "frame",
    "max_samples": 96,
    "grad_accumulation_steps": 16,
    "max_grad_norm": 0.3,
    "epochs": 200,
    "num_warmup_updates": 3000,
    "save_per_updates": 10000,
    "keep_last_n_checkpoints": -1,
    "last_per_updates": 5000,
    "finetune": true,
    "file_checkpoint_train": "",
    "tokenizer_type": "pinyin",
    "tokenizer_file": "",
    "mixed_precision": "bf16",
    "logger": "tensorboard",
    "bnb_optimizer": true
}

Last day:

{
    "exp_name": "F5TTS_v1_Base",
    "learning_rate": 0.00005,
    "batch_size_per_gpu": 2000,
    "batch_size_type": "frame",
    "max_samples": 96,
    "grad_accumulation_steps": 16,
    "max_grad_norm": 0.3,
    "epochs": 200,
    "num_warmup_updates": 500,
    "save_per_updates": 5000,
    "keep_last_n_checkpoints": -1,
    "last_per_updates": 5000,
    "finetune": true,
    "file_checkpoint_train": "",
    "tokenizer_type": "pinyin",
    "tokenizer_file": "",
    "mixed_precision": "bf16",
    "logger": "tensorboard",
    "bnb_optimizer": true
}

**Reply by AsmoKoskinen** [reply]
I will upload files to the HF, and write a pull request.

**Reply by AsmoKoskinen** [reply]
I have added new files to the HF. But I don't think I'll make a new pull request. There is already a link to the Finnish model in HF.
